backend/Squirrel.Core/Squirrel.Core.BLL/Services/HttpInternalService.cs

```diff
@@ -0,0 +1,34 @@
+﻿using Squirrel.Core.BLL.Interfaces;
+using System.Net.Http.Json;
+
+namespace Squirrel.Core.BLL.Services;
+
+public class HttpInternalService : IHttpInternalService
```

Comment:
probably it can be `HttpClientService`? 

something as you did for FE

and about converting response content to model it can be like:

`    public static class ResponseHelper
    {
        public static async Task<T> GetModelAsync<T>(this HttpResponseMessage response)
        {
            return JsonConvert.DeserializeObject<T>(await response.Content.ReadAsStringAsync());
        }
    }`
    
    
    or separate method here
    
    and base request as 
    
    `public async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request)
    => await _httpClient.SendAsync(request).ConfigureAwait(false);`
    
    and do we need Dispose method here (implement IDisposable)?

Reply:
Thank you very much for your suggested improvements. Regarding IDisposable, after you pointed this out I did some research and, as stated in Microsoft documentation, it's better to use a single instance of HttpClient if possible. Therefore, I have changed HttpClientService to a Singleton, so now there is no need to implement IDisposable.